c/certificateService.c

```diff
@@ -118,36 +119,87 @@ static void respondWithBadRequest(HttpResponse *response) {
 
     jsonStart(p);
     {
-      jsonAddString(p, "error", "The length of the certificate is longer than 4096 bytes");
+      jsonAddString(p, "error", errorMessage);
     }
     jsonEnd(p);
 
     finishResponse(response);
 }
 
+
 static int serveMappingService(HttpService *service, HttpResponse *response)
 {
   HttpRequest *request = response->request;
 
-  if (!strcmp(request->method, methodPOST))
-  {
-    RUsermapParamList *userMapCertificateStructure 
-      = (RUsermapParamList*)safeMalloc31(sizeof(RUsermapParamList),"RUsermapParamList");
-    memset(userMapCertificateStructure, 0, sizeof(RUsermapParamList));
+  if (!strcmp(request->method, methodPOST)) {
+    RUsermapParamList userMapStructure = {0};
```

Comment:
There is an advantage: you don't have to manage this memory; considering how many exits points there are and the current design, that would have been error prone. For 64-bit this doesn't work of course.

I suggest the following structure:

```c
static int serveMappingService(HttpService *service, HttpResponse *response) {
  ALLOC_STRUCT31(...);

  if (error) {
    respondWithBadRequest(...);
    goto out;
  }
 
  if (anotherError) {
    respondWithBadRequest(...);
    goto out;
  } 
  
out:
  FREE_STRUCT31(...);
  return 0;
}

That way it's easier to reason what gets released when, and harder to miss a "free" call.

Reply:
`serveMappingService` could also be refactored not to have to have so many exits points, e.g. by splitting into separate functions (parm validation, RACF parm initialization, the RACF call itself).

Reply:
can we make these change in another PR? e.g. PR546.

Reply:
I am asking because 10.1. is code freeze, we can either make these changes into upcoming release or leave it for the next release, but the bug in the 64-bit version would still be there. Do you think it will decrease the quality of the code?